test/server/cards/01_SOR/units/GeneralTaggeConcernedCommander.spec.ts

```diff
@@ -0,0 +1,78 @@
+describe('General Tagge, Concerned Commander', function () {
+    integration(function (contextRef) {
+        describe('General Tagge\'s ability', function () {
+            beforeEach(function () {
+                contextRef.setupTest({
+                    phase: 'action',
+                    player1: {
+                        hand: ['general-tagge#concerned-commander'],
+                        groundArena: ['vanguard-infantry', 'mercenary-company', 'battlefield-marine', 'wampa'],
+                    },
+                    player2: {
+                        groundArena: ['volunteer-soldier'],
+                    }
+                });
+            });
+
+            it('should give experience token to up to 3 Trooper unit', function () {
+                const { context } = contextRef;
+
+                context.player1.clickCard(context.generalTagge);
+
+                // should be able to select all trooper
+                expect(context.player1).toBeAbleToSelectExactly([context.vanguardInfantry, context.mercenaryCompany, context.battlefieldMarine, context.volunteerSoldier]);
+                context.player1.clickCard(context.vanguardInfantry);
+                context.player1.clickCard(context.mercenaryCompany);
+                context.player1.clickCard(context.battlefieldMarine);
```

Comment:
Can you add a click on the fourth trooper just to confirm it doesn't work (use `clickCardNonChecking()`)